src/mongo/commands.ts

```diff
@@ -34,15 +34,19 @@ export class MongoCommands {
 				const db = await database.getDb();
 				let node = new MongoCollectionNode(db.collection(command.collection), database, command.arguments);
 				await node.getChildren();
-				await editor.showDocument(node, extensionPath, 'cosmos-result.json');
+				await editor.showDocument(node, 'cosmos-result.json');
 				return command;
 			}
 			const result = await database.executeCommand(command);
 			const parsed = JSON.parse(result);
 			if (command.name === 'findOne') {
 				const db = await database.getDb();
-				let node = new MongoDocumentNode(parsed._id, null, parsed);
-				await editor.showDocument(node, extensionPath, 'cosmos-result.json');
+				let id = parsed && parsed._id;
+				if (typeof id === "undefined") {
```

Comment:
Similar to the other case, it seems bug-prone to fix this here rather than where you're using id. (I assume you did this because of `findDocById`? You should just do `child && child.id.toString() === id`)

Reply:
I did this because `JSON.parse(null)` works, while `JSON.parse(undefined)` doesn't